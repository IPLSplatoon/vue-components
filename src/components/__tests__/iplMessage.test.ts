import IplMessage from '../iplMessage.vue';
import { config, mount } from '@vue/test-utils';
import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome';

describe('IplMessage', () => {
    config.global.stubs = {
        FontAwesomeIcon: true
    };

    it('matches snapshot', () => {
        const wrapper = mount(IplMessage, { props: { type: 'info' } });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('matches snapshot when closeable', () => {
        const wrapper = mount(IplMessage, { props: { type: 'info', closeable: true } });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('emits event on close button click', () => {
        const wrapper = mount(IplMessage, { props: { type: 'info', closeable: true } });

        wrapper.getComponent<typeof FontAwesomeIcon>('.close-button').vm.$emit('click');

        expect(wrapper.emitted('close')?.length).toEqual(1);
    });

    it('has expected wrapper class and icon if type is error', () => {
        const wrapper = mount(IplMessage, { props: { type: 'error' } });

        expect(wrapper.getComponent('.icon').attributes().icon).toEqual('exclamation-circle');
        expect(wrapper.get('.ipl-message__wrapper').classes()).toContain('ipl-message__type-error');
    });

    it('has expected wrapper class and icon if type is info', () => {
        const wrapper = mount(IplMessage, { props: { type: 'info' } });

        expect(wrapper.getComponent('.icon').attributes().icon).toEqual('info-circle');
        expect(wrapper.get('.ipl-message__wrapper').classes()).toContain('ipl-message__type-info');
    });

    it('has expected wrapper class and icon if type is warning', () => {
        const wrapper = mount(IplMessage, { props: { type: 'warning' } });

        expect(wrapper.getComponent('.icon').attributes().icon).toEqual('exclamation-triangle');
        expect(wrapper.get('.ipl-message__wrapper').classes()).toContain('ipl-message__type-warning');
    });

    describe('validator: type', () => {
        const validator = IplMessage.props.type.validator as (value?: string | null) => boolean;

        it('allows valid types', () => {
            expect(validator('info')).toEqual(true);
            expect(validator('warning')).toEqual(true);
            expect(validator('error')).toEqual(true);
        });

        it('does not allow unknown types', () => {
            expect(validator('some message')).toEqual(false);
            expect(validator(null)).toEqual(false);
            expect(validator(undefined)).toEqual(false);
        });
    });
});
