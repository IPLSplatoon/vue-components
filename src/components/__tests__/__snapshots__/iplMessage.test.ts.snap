// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IplMessage matches snapshot 1`] = `
"<div class="ipl-message__wrapper ipl-message__type-info">
  <font-awesome-icon-stub icon="info-circle" border="false" fixedwidth="false" flip="false" listitem="false" pulse="false" swapopacity="false" spin="false" symbol="false" inverse="false" bounce="false" shake="false" beat="false" fade="false" beatfade="false" flash="false" spinpulse="false" spinreverse="false" class="icon"></font-awesome-icon-stub>
  <div class="message-content">
    <!--v-if-->
  </div>
</div>"
`;

exports[`IplMessage matches snapshot when closeable 1`] = `
"<div class="ipl-message__wrapper ipl-message__type-info">
  <font-awesome-icon-stub icon="info-circle" border="false" fixedwidth="false" flip="false" listitem="false" pulse="false" swapopacity="false" spin="false" symbol="false" inverse="false" bounce="false" shake="false" beat="false" fade="false" beatfade="false" flash="false" spinpulse="false" spinreverse="false" class="icon"></font-awesome-icon-stub>
  <div class="message-content">
    <ipl-button-stub icon="times" color="transparent" disabled="false" async="false" progressmessage="Working..." successmessage="Done!" requiresconfirmation="false" shortconfirmationmessage="false" small="false" disableonsuccess="false" inline="true" class="close-button"></ipl-button-stub>
  </div>
</div>"
`;
